example/kokkos_pi/kokkos_pi.cpp

```diff
@@ -0,0 +1,381 @@
+  MeshBlock *firstBlock = nullptr;
+  MeshBlock *lastBlock = nullptr;
+
+  // compute an offset due to ghost cells
+  double delta = dxyzCell * static_cast<Real>(NG);
+
+  int idx = 0; // an index into Block coordinate array
+  for (int k_mesh = 0; k_mesh < n_mesh; k_mesh++) {
+    for (int j_mesh = 0; j_mesh < n_mesh; j_mesh++) {
+      for (int i_mesh = 0; i_mesh < n_mesh; i_mesh++, idx++) {
+        // get a new meshblock and insert into chain
+        auto *pmb = new MeshBlock(n_block, 3);
+        if (lastBlock) {
+          lastBlock->next = pmb;
+          pmb->prev = lastBlock;
+        } else {
+          firstBlock = pmb;
+        }
+        // set coordinates of first cell center
+        h_xyz(0, idx) = dxyzCell * (static_cast<Real>(i_mesh * n_block) + 0.5) - delta;
+        h_xyz(1, idx) = dxyzCell * (static_cast<Real>(j_mesh * n_block) + 0.5) - delta;
+        h_xyz(2, idx) = dxyzCell * (static_cast<Real>(k_mesh * n_block) + 0.5) - delta;
+        // Add variable for in_or_out
+        Container<Real> &base = pmb->real_containers.Get();
+        base.setBlock(pmb);
+        base.Add("in_or_out", myMetadata);
+        // repoint lastBlock for next iteration
+        lastBlock = pmb;
+      }
+    }
+  }
+  // copy our coordinates over to Device and wait for completion
+  Kokkos::deep_copy(xyz, h_xyz);
+  Kokkos::fence();
+
+  return firstBlock;
+}
+
+void deleteMesh(MeshBlock *firstBlock) {
+  // deletes the mesh chain starting with firstBlock
+  auto pmb = firstBlock;
+  while (pmb) {
+    auto next = pmb->next;
+    delete pmb;
+    pmb = next;
+  }
+}
+
+result_t naiveKokkos(int n_block, int n_mesh, int n_iter, double radius) {
+  // creates a mesh and rusn a basic Kokkos implementation for looping through blocks.
+
+  // Setup auxilliary variables
+  const int n_block2 = n_block * n_block;
+  const int n_block3 = n_block * n_block * n_block;
+  const int n_mesh3 = n_mesh * n_mesh * n_mesh;
+  const double radius2 = radius * radius;
+  const double radius3 = radius * radius * radius;
+  const Real dxyzCell = radius / static_cast<Real>(n_mesh * n_block);
+  const Real dVol = radius3 / static_cast<Real>(n_mesh3) / static_cast<Real>(n_block3);
+
+  // allocate space for origin coordinates and set up the mesh
+  View2D xyz("xyzBlocks", 3, n_mesh3);
+  MeshBlock *firstBlock = setupMesh(n_block, n_mesh, radius, xyz);
+
+  // first A  naive Kokkos loop over the mesh
+  // This policy is over one block
+  auto policyBlock = Kokkos::RangePolicy<>(Kokkos::DefaultExecutionSpace(), 0, n_block3,
+                                           Kokkos::ChunkSize(512));
+
+  MeshBlock *pStart = firstBlock;
+  double time_basic = kernel_timer_wrapper(0, n_iter, [&]() {
+    MeshBlock *pmb = pStart;
+    for (int iMesh = 0; iMesh < n_mesh3; iMesh++, pmb = pmb->next) {
+      Container<Real> &base = pmb->real_containers.Get();
+      auto inOrOut = base.PackVariables({Metadata::Independent});
```

Comment:
This will be cached on a per meshblock basis.

Reply:
Right, so when we come in for the second iteration, we get the cached version, correct?  That is why I put the burn iterations for the timer as 0.

Reply:
Yep that's right :)